kiosk/default.nix

```diff
@@ -11,6 +11,12 @@ python3Packages.buildPythonApplication rec {
     (path: type: type != "directory" ||  baseNameOf path != "venv")
     ./.;
 
+  postPatch = ''
+    substituteInPlace kiosk_browser/system.py \
+      --replace @name@ ${system_name} \
```

Comment:
Assumes `system_name` never includes whitespace.

Reply:
Well spotted!

I considered using `substituteAllInPlace`:

```diff
diff --git a/kiosk/default.nix b/kiosk/default.nix
index 811797e8..3e0dba30 100644
--- a/kiosk/default.nix
+++ b/kiosk/default.nix
@@ -6,15 +6,14 @@ let qtx=qt5; in
 python3Packages.buildPythonApplication rec {
   pname = "playos_kiosk_browser";
   version = "0.1.0";
+  inherit system_name system_version;
 
   src = builtins.filterSource
     (path: type: type != "directory" ||  baseNameOf path != "venv")
     ./.;
 
   postPatch = ''
-    substituteInPlace kiosk_browser/system.py \
-      --replace @name@ ${system_name} \
-      --replace @version@ ${system_version}
+    substituteAllInPlace kiosk_browser/system.py
   '';
 
   doCheck = false;
diff --git a/kiosk/kiosk_browser/system.py b/kiosk/kiosk_browser/system.py
index 472a5445..f5386ade 100644
--- a/kiosk/kiosk_browser/system.py
+++ b/kiosk/kiosk_browser/system.py
@@ -1,2 +1,2 @@
-NAME = "@name@"
-VERSION = "@version@"
+NAME = "@system_name@"
+VERSION = "@system_version@"
```

But that’s brittle, someone could remove the `inherits` without noticing and we would loose the substitution. So, I sticked to `substituteInPlace`.